week-4/E-arrays-of-objects/exercise-3.js

```diff
@@ -59,14 +59,32 @@ var restaurantFinderApplication = {
   applicationVersion: "1.0",
   restaurants: restaurants,
   findAvailableRestaurants: function(numberOfPeople) {
+    function findRestaurant(restaurant) {
+      return (
+        restaurant.totalSeats - restaurant.numberOfCustomers > numberOfPeople
+      );
+    }
+    return restaurants.filter(findRestaurant).map(R => R.name);
+
     // Complete here
   },
   findRestaurantServingDish: function(dishName) {
+    function checkDish(dish) {
+      return restaurants.includes(dish);
+    }
+    return restaurants.filter(checkDish).map(r => r.name);
     // Complete here
   },
   countNumberOfRestaurantsInArea: function(area) {
-    // Complete here
+    function adressCount(address1, address2) {
```

Comment:
You never use area in here, which should make you suspicious whether this works. (If you change it to west instead of center, it still returns 2).
As before you need a filter which checks if the restaurant is what you're looking for - that `restaurant.address.area` is the same as the `area` parameter - and then you don't need a map.